File: ruby/g10.parent/Student.rb
#!/usr/bin/ruby
# class declaration
class Student < Person
  ###### Initializer ######
  def initialize(name, gpa=0.0) # constructor
    setName(name)      # call mutator
    setGPA(gpa)        # call mutator
  end

  ###### Mutators ######
  def setGPA (gpa=0.0)
    @gpa = gpa         # set instance data
  end
  
  ###### Accessors ######
  def getGPA
    @gpa               # return instance data
  end
end

File: python/g10.parent/Student.py
#!/usr/bin/python
from Person import Person    # include Person.py

# class declaration
class Student(Person):
  ###### INITIALIZER ######
  def __init__(self, name, gpa=0.0):
    self.name = name  # call parent's mutator
    self.gpa = gpa    # call mutator

  ###### PROPERTIES ######
  @property
  def name (self):
    return self._gpa

  @name.setter
  def name (self, gpa):
    self._gpa = gpa
